src/store/attachment.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * @copyright Copyright (c) 2020 Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @author Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { AttachmentApi } from './../services/AttachmentApi'
+import Vue from 'vue'
+
+const apiClient = new AttachmentApi()
+
+export default {
+	state: {
+		attachments: {},
+	},
+	getters: {
+		attachmentsByCard: state => (cardId) => {
+			if (typeof state.attachments[cardId] === 'undefined') {
+				return []
+			}
+			return state.attachments[cardId]
+		},
+
+	},
+	mutations: {
+		createAttachment(state, { cardId, attachment }) {
+			if (typeof state.attachments[cardId] === 'undefined') {
+				Vue.set(state.attachments, cardId, attachment)
```

Comment:
This will cause the attachments array be no array any longer. For applying full list on load we have createAttachments.

Reply:
So, I can remove the lines 43 - 45, right?

Reply:
No, we need one muation to add an array of attachments and one to add a single attachment. But the single attachment one (L42) should also handle that the array doesn't exist yet in the store, which is what this if statement is used for.